packages/shell-api/src/shell-internal-state.ts

```diff
@@ -57,7 +58,10 @@ export default class ShellInternalState {
   async fetchConnectionInfo(): Promise<void> {
     if (!this.cliOptions.nodb) {
       this.connectionInfo = await this.currentDb._mongo._serviceProvider.getConnectionInfo();
-      this.messageBus.emit('mongosh:connect', this.connectionInfo.extraInfo);
+      this.messageBus.emit('mongosh:connect', {
+        ...this.connectionInfo.extraInfo,
+        uri: redactInfo(this.connectionInfo.extraInfo.uri)
```

Comment:
How is this URI used? I think the regex that would catch it would always change it to `<mongodb uri>`
https://github.com/mongodb-js/redact/blob/master/lib/regexes.js#L26

Reply:
The URI is used when we make a second connection for auth, since we cannot reconnect the same MongoClient with new authentication. The messageBus is only used for logging, so we don't really care that the URI is logged or not. In this case I'm just overwriting it with `redact`, but we could delete it from the object with a couple extra lines of code.  The redact should remove a little more selectively, for example `"mongodb://<ip address>:27017"` for `:"mongodb://localhost:27017"

Reply:
Ah cool.
Gave it a quick test - I think the redact might do some things we don't quite expect sometimes hah
```js
const redactInfo = require('mongodb-redact');

console.log(redactInfo('mongodb://localhost:27017'));
// outputs '<mongodb uri>'

console.log(redactInfo('mongodb+srv://testUserForTesting:notMyRealPassword@some-atlas-db.mongodb.net/test'));
// outputs 'mongodb+srv://testUserForTesting:<email>/test'
```
Not much to do here I suppose but something to keep in mind for the redaction package.

Reply:
Errrr thinking about this more.... Maybe our redaction package regexes could be not work properly with symbols in the password or username for connection strings and not redact properly. 🤔

Reply:
I think when you have a "full" URI (meaning the generated one from the CLI options) `"mongodb://127.0.0.1:27017"` --> `"mongodb://<ip address>:27017"` it works slightly better. I do agree there's improvements to be made and I think we should address that ticket and the logging one ASAP